### PR TITLE
ref(e2e): Fetch received event in the tested app

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Enhancement

## :scroll: Description

This PR prepares E2E test for switch to Maestro.

Where it's better to fetch the event in the simulator, as maestro test are declarative.

## :green_heart: How did you test it?
ci, run locally

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

#skip-changelog

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
Let's wait for RN 0.76.1 to fix CI builds. So we can merge this without https://github.com/getsentry/sentry-react-native/pull/4209/commits/5555f27c79f44e44e133090c05ef260d56a34378

File: dev-packages/e2e-tests/package.json [hunk 1]
```diff
@@ -23,12 +23,16 @@
...
     "typescript": "4.9.5",
     "webdriverio": "^8.27.0"
   },
   "dependencies": {
     "minimist": "1.2.8",
     "semver": "7.6.3",
     "xcode": "3.0.1"
+  },
+  "peerDependencies": {
+    "react-native-launch-arguments": "^4.0.2"
```
Comment (lucas-zimerman) [inline dev-packages/e2e-tests/package.json:36]: Q: adding it here and on peerDependencies? with that we got a reference to react-native-launch-arguments on the root yarn.lock

**Review by lucas-zimerman — APPROVED** [review]
Overall looks ok, just a minor question in regard to the package usage, LGTM!

File: dev-packages/e2e-tests/package.json (hunk 1 above)
Comment (krystofwoldrich) [reply dev-packages/e2e-tests/package.json:36]: The reason to add `react-native-launch-arguments` to peer dependencies is to indicate that the e2e-component package need this to be installed along side.

To be absolutely exact we should also include @sentry/react-native in here.

Comment (krystofwoldrich) [reply dev-packages/e2e-tests/package.json:36]: In the root yarn.lock the dep is already based on the devDependencies.

**Pull request merged**